src/taipy/gui/gui_actions.py

```diff
@@ -195,3 +196,74 @@ def invoke_callback(gui: Gui, state_id: str, callback: t.Callable, args: t.Union
 def invoke_state_callback(gui: Gui, state_id: str, callback: t.Callable, args: t.Union[t.Tuple, t.List]) -> t.Any:
     warnings.warn("'invoke_state_callback()' was deprecated in Taipy GUI 2.0. Use 'invoke_callback()' instead.")
     return invoke_callback(gui, state_id, callback, args)
+
+
+def invoke_long_callback(
+    state: State,
+    user_function: t.Callable,
+    user_function_args: t.Union[t.Tuple, t.List] = [],
+    user_status_function: t.Optional[t.Callable] = None,
+    user_status_function_args: t.Union[t.Tuple, t.List] = [],
+    period=0,
+):
+    """Invoke a long running user callback (uses Threads).
+
+    See the [User Manual section on Long Running Callbacks](../../gui/callbacks/#long-running-callbacks)
+    for details on when and how this function can be used.
+
+    Arguments:
+        state (State^): The `State^` instance, which is an argument of the callback function.
+        user_function (Callable[[...], None]): The function that will be run independently of taipy gui (cannot use `State^`).
+        user_function_args (Optional[List|Tuple]): The remaining arguments, as a List.
+        user_status_function (Optional(Callable[[State^, bool, ...], None])): The optional user-defined status function that is invoked at the end (of and possibly during) the runtime of the user_function.<br/>
+            The first parameter of this function **must** be a `State^`. The second parameter of this function **must** be a Union[`bool`, `int`] that will indicate a progression (`int`) or the final status (`bool`)
+        user_status_function_args (Optional[List|Tuple]): The remaining arguments ot the user status function, as a List.
+        period (int): The interval in milli-seconds at which the user_status_function is called (default value is 0).</br>
+            When set to a value < 500, the user_status_function is called only when the user_function is terminated.
+    """
+    if state and isinstance(state._gui, Gui):
```

Comment:
if not state or not isinstance(state._gui, Gui):
   warnings.warn("'invoke_long_callback()' must be called in the context of a callback.")
   return

You get what I mean :D

Reply:
Less visible complexity...
Yes...